File: jugglerhelper/src/main/java/net/maxsmr/jugglerhelper/fragments/toolbar/BaseCustomJugglerToolbarFragment.java
package net.maxsmr.jugglerhelper.fragments.toolbar;

import android.graphics.drawable.Drawable;
import android.support.annotation.IdRes;

import org.jetbrains.annotations.Nullable;

import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import net.maxsmr.jugglerhelper.R;

public abstract class BaseCustomJugglerToolbarFragment extends BaseJugglerToolbarFragment {

    @IdRes
    protected abstract int getToolbarTitleId();

    @IdRes
    protected abstract int getToolbarLogoId();

    @Override
    public void setTitle(@Nullable CharSequence title) {
        View rootView = getView();
        if (rootView == null) {
            throw new IllegalStateException(getClass().getSimpleName() + " is not attached to activity");
        }
        TextView titleView = rootView.findViewById(getToolbarTitleId());
        if (titleView != null) {
            super.setTitle(getString(R.string.no_data));
            titleView.setText(title);
            titleView.setVisibility(TextUtils.isEmpty(title) ? View.GONE : View.VISIBLE);
        } else {
            super.setTitle(title);
        }
    }

    @Override
    public void setLogo(@Nullable Drawable logo) {
        View rootView = getView();
        if (rootView == null) {
            throw new IllegalStateException(getClass().getSimpleName() + " is not attached to activity");
        }
        ImageView logoView = rootView.findViewById(getToolbarLogoId());
        if (logoView != null) {
            super.setLogo(null);
            logoView.setImageDrawable(logo);
            logoView.setVisibility(logo == null ? View.GONE : View.VISIBLE);
        } else {
            super.setLogo(logo);
        }
    }
}

File: jugglerhelper/src/main/kotlin/net/maxsmr/jugglerhelper/fragments/toolbar/BaseCustomJugglerToolbarFragment.kt
package net.maxsmr.jugglerhelper.fragments.toolbar

import android.graphics.drawable.Drawable
import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.IdRes

abstract class BaseCustomJugglerToolbarFragment : BaseJugglerToolbarFragment() {

    @IdRes
    protected open val toolbarTitleId: Int = 0

    @IdRes
    protected open val toolbarLogoId: Int = 0

    override fun applyTitle(title: CharSequence?) {
        with(toolbarTitleId) {
            val titleView = if (this != 0) view?.findViewById<TextView>(this) else null
            if (titleView != null) {
                super.applyTitle("")
                titleView.text = title
                titleView.visibility = if (TextUtils.isEmpty(title)) View.GONE else View.VISIBLE
            } else {
                super.applyTitle(title)
            }
        }
    }

    override fun applyLogo(logo: Drawable?) {
        with(toolbarLogoId) {
            val logoView = if (this != 0) view?.findViewById<ImageView>(this) else null
            if (logoView != null) {
                super.applyLogo(null)
                logoView.setImageDrawable(logo)
                logoView.visibility = if (logo == null) View.GONE else View.VISIBLE
            } else {
                super.applyLogo(logo)
            }
        }
    }
}
